client/src/controllers/BulkController.test.js

```diff
@@ -0,0 +1,92 @@
+import { Application } from '@hotwired/stimulus';
+import { BulkController } from './BulkController';
+
+describe('BulkController', () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+    <div data-controller="w-bulk">
+      <input type="checkbox" data-w-bulk-target="all">
+      <div>
+        <input type="checkbox" data-w-bulk-target="item" disabled>
+        <input type="checkbox" data-w-bulk-target="item">
+        <input type="checkbox" data-w-bulk-target="item">
+      </div>
```

Comment:
These tests are a great start but you are actually only testing that if the checkboxes change, they will reflect that change.

What is missing is the usage of `data-action` so that changing the select all checkbox triggers a change in the others. We also need a `data-action` on each individual checkbox item so that it will also be able to potentially trigger the change to the select all checkbox.

So `data-...target` for an element we want to modify and `data-action` for something we want to add an event listener to. In this case we need both.


```suggestion
      <input type="checkbox" data-w-bulk-target="all" data-action="w-bulk#toggleAll">
      <div>
        <input type="checkbox" data-w-bulk-target="item" disabled data-action="w-bulk#toggle">
        <input type="checkbox" data-w-bulk-target="item" data-action="w-bulk#toggle">
        <input type="checkbox" data-w-bulk-target="item" data-action="w-bulk#toggle">
      </div>
```

Reply:
Okay sure